app/plugin_key_value_store.go

```diff
@@ -81,17 +98,17 @@ func (a *App) CompareAndDeletePluginKey(pluginId string, key string, oldValue []
 func (a *App) GetPluginKey(pluginId string, key string) ([]byte, *model.AppError) {
 	if kv, err := a.Srv().Store.Plugin().Get(pluginId, key); err == nil {
 		return kv.Value, nil
-	} else if err.StatusCode != http.StatusNotFound {
+	} else if _, ok := err.(*store.ErrNotFound); !ok {
```

Comment:
Let's use `errors.As` here

Reply:
I knew you could ask for that :) but what I don't like about that approach here is it forces us to create a pointer just to assert the error is of that type, but we won't use that pointer for anything else. Different is when we use this approach in the `switch` sentence because there we do want to use the pointed error. WDYT?

Reply:
The API is designed in a way that forces us to create that pointer. But not using `errors.As` makes the code fragile and prone to breakages. If you don't want to create a pointer as a separate variable, you can do

```
if nfErr := new(store.ErrNotFound); errors.As(err, &nfErr) {
}
```

Reply:
Ok I will make that change, but I think you meant:

```go
if nfErr := new(store.ErrNotFound); !errors.As(err, &nfErr) {
}
```

Because we are checking that the error is **not** `store.ErrNotFound`. BTW that is why I'm returning 500 insted of 404 :)

Let me check the old code again because I'm hesitating now

Reply:
Ohhh that was the reason. My apologies, I couldn't understand what you meant by "distinct of ErrNotFound". Yes you are right of course. I failed to see it. No need to change to 404. Only the `errors.As` change is needed.

Reply:
:)